src/serac/numerics/functional/tensor.hpp

```diff
@@ -358,6 +358,22 @@ SERAC_HOST_DEVICE constexpr auto operator*(T /*other*/, zero)
   return zero{};
 }
 
+/** @brief the quotient of `zero` with something else is always `zero` */
+template <typename T>
+SERAC_HOST_DEVICE constexpr auto operator/(zero, T /*other*/)
+{
+  static_assert(!is_zero<T>::value, "Zero divided by zero is undefined.");
+  return zero{};
+}
+
+/** @brief the quotient of something else with `zero` is undefined */
+template <typename T>
+SERAC_HOST_DEVICE constexpr auto operator/(T /*other*/, zero)
+{
+  SLIC_ASSERT("Division by zero is undefined.");
```

Comment:
does `SLIC_ASSERT` produce a runtime error? We know at compile time that `T{} / zero{}` is not defined, so we could catch these mistakes before running the program.

Reply:
Yes - I was going to ask about this. When I tried to do a `static_assert` that was always false, the compiler issued an error. Is there a better way to do this?

Reply:
here's one way: https://godbolt.org/z/Wefhn8Pnc

another is to just not define the operation where `serac::zero` is the denominator, and it will fail to compile (although the error message may still be confusing)

Reply:
oops, I guess we can't do the second suggestion, since it'll be captured by your `operator/(zero, T)`